src/python/pants/backend/java/compile/javac.py

```diff
@@ -5,47 +5,31 @@
 
 import logging
 from dataclasses import dataclass
+from itertools import chain
 
 from pants.backend.java.compile.javac_binary import JavacBinary
 from pants.backend.java.target_types import JavaSources
 from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.addresses import Addresses
 from pants.engine.fs import EMPTY_DIGEST, AddPrefix, Digest, MergeDigests, RemovePrefix
 from pants.engine.process import BashBinary, Process, ProcessResult
 from pants.engine.rules import Get, MultiGet, collect_rules, rule
-from pants.engine.target import Dependencies, DependenciesRequest, Sources, Target, Targets
+from pants.engine.target import CoarsenedTarget, CoarsenedTargets, Sources, Targets
 from pants.jvm.resolve.coursier_fetch import (
     CoursierLockfileForTargetRequest,
     CoursierResolvedLockfile,
     MaterializedClasspath,
     MaterializedClasspathRequest,
 )
 from pants.jvm.resolve.coursier_setup import Coursier
-from pants.option.subsystem import Subsystem
 from pants.util.logging import LogLevel
 
 logger = logging.getLogger(__name__)
 
 
-class JavacSubsystem(Subsystem):
```

Comment:
Note to reviewers: it looks like due to branch juggling I failed to remove this in #12283.  I can split the removal into its own PR if desired, but it just boils down to this trivial deletion of an unused class